java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink.java

```diff
@@ -199,15 +200,29 @@ private Map<String, Object> buildDoc(SinkRow row)
             var type = columnDescs.get(i).getDataType().getTypeName();
             Object col = row.get(i);
             switch (type) {
-                case DATE:
-                case TIME:
-                case TIMESTAMP:
-                case TIMESTAMPTZ:
                     // es client doesn't natively support java.sql.Timestamp/Time/Date
                     // so we need to convert Date/Time/Timestamp type into a string as suggested in
                     // https://github.com/elastic/elasticsearch/issues/31377#issuecomment-398102292
+                case DATE:
                     col = col.toString();
                     break;
+                    // construct java.sql.Time/Timestamp with milliseconds time value.
+                    // it will use system timezone by default, so we have to set timezone manually
+                case TIME:
+                    SimpleDateFormat tDfm = new SimpleDateFormat("HH:mm:ss.SSS");
```

Comment:
The formatter can be stored as a static field to avoid repeatedly creating the objects.

Reply:
java.util.Date and SimpleDateFormatter) aren’t thread-safe. Using a static formatter seems to lead to incorrect data and test failure. https://buildkite.com/risingwavelabs/pull-request/builds/36192#018c1499-1f24-4950-a41a-d703aaaec4e2

Reply:
That's one of the many reasons to prefer `java.time.*` But yes it may need a prior refactor for other remote sinks as well.